lti_consumer/tests/unit/plugin/test_views_lti_nrps.py

```diff
@@ -0,0 +1,317 @@
+            rel_part = rel_part.replace('rel="', '').replace('"', '').strip()
+            result[rel_part] = link_part
+        return result
+
+
+class LtiNrpsContextMembershipViewsetTestCase(LtiNrpsTestCase):
+    """
+    Test LTI-NRPS Context Membership Endpoint
+    """
+
+    def test_unauthenticated_request(self):
+        """
+        Test if context membership throws 403 if request is unauthenticated
+        """
+        response = self.client.get(self.context_membership_endpoint)
+        self.assertEqual(response.status_code, 403)
+
+    def test_token_with_incorrect_scope(self):
+        """
+        Test if context membership throws 403 if token don't have correct scope
+        """
+        self._set_lti_token()
+        response = self.client.get(self.context_membership_endpoint)
+        self.assertEqual(response.status_code, 403)
+
+    @patch('lti_consumer.plugin.views.expose_pii_fields', return_value=False)
+    @patch(
+        'lti_consumer.plugin.views.compat.get_course_members',
+        side_effect=patch_get_memberships()
+    )
+    def test_token_with_correct_scope(self, get_course_members_patcher, expose_pii_fields_patcher):  # pylint: disable=unused-argument
+        """
+        Test if context membership returns correct response when token has correct scope
+        """
+        self._set_lti_token('https://purl.imsglobal.org/spec/lti-nrps/scope/contextmembership.readonly')
+        response = self.client.get(self.context_membership_endpoint)
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response['content-type'], 'application/vnd.ims.lti-nrps.v2.membershipcontainer+json')
+
+    @patch('lti_consumer.plugin.views.expose_pii_fields', return_value=False)
+    @patch(
+        'lti_consumer.plugin.views.compat.get_course_members',
+        side_effect=patch_get_memberships({
+            'student': 4
+        })
+    )
+    def test_get_without_pii(self, get_course_members_patcher, expose_pii_fields_patcher):  # pylint: disable=unused-argument
+        """
+        Test context membership endpoint response structure with PII not exposed.
+        """
+        self._set_lti_token('https://purl.imsglobal.org/spec/lti-nrps/scope/contextmembership.readonly')
+        response = self.client.get(self.context_membership_endpoint)
+        self.assertEqual(response.data['id'], 'http://testserver{}'.format(self.context_membership_endpoint))
+        self.assertEqual(len(response.data['members']), 4)
+        self.assertEqual(response.has_header('Link'), False)
+
+        expose_pii_fields_patcher.assert_called()
+
+        # name & email should not be exposed.
+        member_fields = response.data['members'][0].keys()
+        self.assertEqual(all([
+            'user_id' in member_fields,
+            'roles' in member_fields,
+            'status' in member_fields,
+            'email' not in member_fields,
+            'name' not in member_fields,
+        ]), True)
```

Comment:
This would be shorter, clearer, and more informative if you wrote out the asserts individually.

Reply:
Updated as requested.